src/app_service/common/json_utils.nim

```diff
@@ -36,15 +36,15 @@ template getProp(obj: JsonNode, prop: string, value: var typedesc[uint64]): bool
 
 template getProp(obj: JsonNode, prop: string, value: var typedesc[string]): bool =
   var success = false
-  if (obj.kind == JObject and obj.contains(prop)):
+  if (obj.kind == JObject and obj.contains(prop) and obj[prop].kind == JString):
```

Comment:
All functions in this file should have this check `and obj[prop].kind == ....` but when I did that I couldn't start the app (got crash) which undoubtedly means that in some places we have incorrect type usage or receiving incorrect json data. That should be fixed in another PR.